src/renderer/components/kube-object-menu/kube-object-menu.tsx

```diff
@@ -53,61 +65,61 @@ export class KubeObjectMenu<T extends KubeObject> extends React.Component<KubeOb
 
   @boundMethod
   async update() {
-    hideDetails();
-    editResourceTab(this.props.object);
+    this.props.hideDetails();
+    this.props.editResourceTab(this.props.object);
   }
 
   @boundMethod
   async remove() {
-    hideDetails();
+    this.props.hideDetails();
     const { object, removeAction } = this.props;
 
     if (removeAction) await removeAction();
     else await this.store.remove(object);
   }
 
   @boundMethod
-  renderRemoveMessage() {
+  getRemoveMessage() {
     const { object } = this.props;
 
     if (!object) {
       return null;
     }
 
+    const breadcrumbParts = [object.getNs(), object.getName()];
+
+    const breadcrumb = breadcrumbParts.filter(identity).join("/");
+
     return (
-      <p>Remove {object.kind} <b>{object.getName()}</b>?</p>
+      <p>
+        Remove {object.kind} <b>{breadcrumb}</b> from <b>{this.props.clusterName}</b>?
+      </p>
     );
   }
 
   getMenuItems(): React.ReactChild[] {
     const { object, toolbar } = this.props;
 
-    if (!object) {
-      return [];
-    }
-
-    return KubeObjectMenuRegistry
-      .getInstance()
-      .getItemsForKind(object.kind, object.apiVersion)
-      .map(({ components: { MenuItem }}, index) => (
-        <MenuItem
-          object={object}
-          key={`menu-item-${index}`}
-          toolbar={toolbar}
-        />
-      ));
+    return this.props.kubeObjectMenuItems.map((MenuItem, index) => (
+      <MenuItem
+        object={object}
+        toolbar={toolbar}
+        // TODO: Fix misuse of index in key
```

Comment:
Isn't this only a misuse if the items will be reordered?

Reply:
I would avoid using index if UID is available as a convention but your argument is valid, so I removed the comment.